### PR TITLE
bump to kube 1.25.2

### PR DESCRIPTION
**Pull request by sanchezl**
- cleanup: remove dev Dockerfile
- bump 1.25.0: EditRegistriesConfig signature changed
-  bump 1.25.0: BuildConfigInterface changed
- bump 1.25.0: handle new podspec.hostusers field
- bump 1.25.0: APIObjectVersioner moved pkg
- bump 1.25.2: go mod vendor
- bump 1.25.2: update go.mod deps
-  bump 1.25.0: use go 1.19

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,65 +3,65 @@ module github.com/openshift/openshift-controller-manager
...
-	k8s.io/kubectl v0.24.0
-	k8s.io/kubernetes v1.24.0
-	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
+	k8s.io/api v0.25.0
+	k8s.io/apimachinery v0.25.0
+	k8s.io/apiserver v0.25.0
+	k8s.io/client-go v0.25.0
+	k8s.io/component-base v0.25.0
+	k8s.io/controller-manager v0.25.0
+	k8s.io/klog/v2 v2.80.1
```
Comment (ingvagabund) [inline go.mod:30]: https://github.com/kubernetes/kubernetes/blob/a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2/go.mod (1.25.0 release) sets k8s.io/klog/v2 to `v2.70.1`

File: go.mod [hunk 2]
```diff
@@ -3,65 +3,65 @@ module github.com/openshift/openshift-controller-manager
...
+	k8s.io/kubectl v0.25.0
+	k8s.io/kubernetes v1.25.0
+	k8s.io/utils v0.0.0-20220823124924-e9cbc92d1a73
```
Comment (ingvagabund) [inline go.mod:33]: https://github.com/kubernetes/kubernetes/blob/a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2/go.mod (1.25.0 release) sets k8s.io/utils to `v0.0.0-20220728103510-ee6ede2d64ed`

**Review by ingvagabund — COMMENTED** [review]
Versions/revisions of some dependencies in go.mod do not correspond to k8s v1.25.0 [go.mod](https://github.com/kubernetes/kubernetes/blob/a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2/go.mod):
- github.com/BurntSushi/toml
- k8s.io/klog/v2 
- k8s.io/utils
- and others

@sanchezl do you think it is worth keeping them in sync with 1.25.0 release?

File: pkg/image/controller/trigger/image_trigger_controller_test.go [hunk 3]
```diff
@@ -153,43 +154,51 @@ type fakeBuildConfigInterface struct {
 	namespace string
 }
 
-func (fakeBuildConfigInterface) Create(ctx context.Context, bc *buildv1.BuildConfig, opts metav1.CreateOptions) (*buildv1.BuildConfig, error) {
+func (*fakeBuildConfigInterface) Apply(context.Context, *buildapply.BuildConfigApplyConfiguration, metav1.ApplyOptions) (*buildv1.BuildConfig, error) {
```
Comment (ingvagabund) [inline pkg/image/controller/trigger/image_trigger_controller_test.go:157]: @sanchezl just a nit: worth keeping the same order of methods to reduce the diff

**Comment by ingvagabund** [comment]
@sanchezl worth updating it to 1.25.2

**Comment by sanchezl** [comment]
> @sanchezl worth updating it to 1.25.2

Done.

File: pkg/image/controller/trigger/image_trigger_controller_test.go (hunk 3 above)
Comment (sanchezl) [reply pkg/image/controller/trigger/image_trigger_controller_test.go:157]: I didn't re-order the methods (just added some), but there were enough changes[^1] to the existing methods that the diff does not match them up.


[^1]: Consistently used pointer receivers. Cleaned up unused parameters names.

File: go.mod (hunk 1 above)
Comment (sanchezl) [reply go.mod:30]: > @sanchezl do you think it is worth keeping them in sync with 1.25.0 release?

No. I just let `go mod` figure it out. I try to minimize explicit pins as much as possible.

**Pull request merged**